### DISCUSSION TITLE
view vlans on specific device

### DISCUSSION THREAD

**Discussion by jms234** [netbox-community/netbox#13252] [Q&A]
Just starting with Netbox and have setup a webhook to allow some of our junior admins to change the vlan an access port is in and it works fine.  

The problem is though that there seems to be no way of knowing whether a vlan is on a specific device or not because vlans are interface specific and not device specific so they could select a vlan which does not exist on that switch and adding it involves more work ie. extending from core/distro etc. 

Is there a way I am missing of being able to see this information ie. vlans on a specific device ? 

The only "solution" I have come up with so far is to use the device comments field to list the vlans on the switch but thought there might be a more elegant solution.

**Comment by adparis99** [comment]
As far as I'm aware, there isn't an official way to do _exactly_ what you mentioned, but VLAN Groups can get you pretty close. These can't be assigned per device, but they can be assigned per rack, location, site, site group, or region. 

As an example, if you have a few devices in a rack named Rack A, you could create a VLAN Group that is scoped to that rack. Then when your junior admins want to adjust the VLAN on an interface from a Rack A device, if they filter by VLAN Group they will see only the VLANs available to that device.

You can also go a step further and add every VLAN in your organization to a group of some kind, the effect being that users won't even need to filter by VLAN Group, as the untagged/tagged VLAN dropdowns will automatically be filtered by what's available to that device.

**Comment by jms234** [comment]
Many thanks for that, I will have to have a look at vlan groups and see how they work if I can ever get past how to add an IP to an interface :)
